tests/test_fixity.py

```diff
@@ -634,3 +636,78 @@ def test_scanall_if_sort_argument_is_passed(
             params={"username": STORAGE_SERVICE_USER, "api_key": STORAGE_SERVICE_KEY},
         ),
     ]
+
+
+@mock.patch("requests.get")
+def test_main_handles_exception_if_environment_key_is_missing(
+    _get: mock.Mock, mock_check_fixity: List[mock.Mock]
+) -> None:
+    _get.side_effect = mock_check_fixity
+
+    response = fixity.main(["scan", str(uuid.uuid4())])
+
+    assert str(response) == "Missing environment variable: STORAGE_SERVICE_URL"
+    assert isinstance(response, ArgumentError)
+
+
+@mock.patch("requests.get")
+def test_scanall_handles_exception_if_storage_service_raises_exception(
+    _get: mock.Mock, environment: None
+) -> None:
+    _get.side_effect = [
+        mock.Mock(
+            **{
+                "status_code": 401,
+            },
+            spec=requests.Response,
+        )
+    ]
+
+    response = fixity.main(["scanall"])
+
+    assert (
+        str(response)
+        == f'Storage service at "{STORAGE_SERVICE_URL}" failed authentication while requesting AIPs'
+    )
+    assert isinstance(response, StorageServiceError)
+
+
+@mock.patch("requests.get")
+def test_main_verifies_urls_with_trailing_slash(
+    _get: mock.Mock,
+    mock_check_fixity: List[mock.Mock],
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    _get.side_effect = [
+        mock.Mock(**{"status_code": 200}, spec=requests.Response),
+        mock.Mock(**{"status_code": 401}, spec=requests.Response),
+    ]
+    aip_id = uuid.uuid4()
+    stream = io.StringIO()
+    ss_url = "http://foo"
+    monkeypatch.setenv("STORAGE_SERVICE_URL", ss_url)
+    monkeypatch.setenv("STORAGE_SERVICE_USER", STORAGE_SERVICE_USER)
+    monkeypatch.setenv("STORAGE_SERVICE_KEY", STORAGE_SERVICE_KEY)
+    report_url = "http://bar"
+    monkeypatch.setenv("REPORT_URL", report_url)
+    monkeypatch.setenv("REPORT_USERNAME", "test")
+    monkeypatch.setenv("REPORT_PASSWORD", "test123")
+
+    response = fixity.main(["scan", str(aip_id)], stream=stream)
+
+    assert response is None
```

Comment:
Notice how this assertion demonstrates that this type annotation is inaccurate:

https://github.com/artefactual/fixity/blob/1387341586837f8a5f44aa43a29662d87ef0f500/fixity/fixity.py#L317

The return value of main should include `None` as well?